src/openrct2/interface/Window.cpp

```diff
@@ -916,6 +916,18 @@ static constexpr float kWindowScrollLocations[][2] = {
         });
     }
 
+    std::vector<Viewport*> WindowGetActiveViewports()
+    {
+        std::vector<Viewport*> viewports;
+        WindowVisitEach([&viewports](WindowBase* w) {
+            if (w->viewport != nullptr && !(w->flags & WF_DEAD))
+            {
+                viewports.push_back(w->viewport);
+            }
```

Comment:
I don't think this ever happens. Consider `WindowManager::Close`:

https://github.com/OpenRCT2/OpenRCT2/blob/c934280589782d16a61d8e249b2a59bdd8a0135b/src/openrct2-ui/WindowManager.cpp#L945-L951

Note that the viewport gets removed before the window is flagged with `WF_DEAD`.